transform/src/main/java/co/cask/wrangler/PreconditionException.java

```diff
@@ -0,0 +1,10 @@
+package co.cask.wrangler;
+
+/**
+ * Thrown when there are issues with pre-conditions.
+ */
+public class PreconditionException extends Exception {
```

Comment:
if this extends IllegalArgumentException instead, will it simplify exception handling?